shell/src/main/java/org/apache/accumulo/shell/commands/ConfigCommand.java

```diff
@@ -353,6 +344,31 @@ public int execute(final String fullCommand, final CommandLine cl, final Shell s
     return 0;
   }
 
+  private void validatePropertyType(String property, String value, String fullCommand,
+      String tableName) throws BadArgumentException {
+    if (tableName != null) {
+      if (!Property.isValidTablePropertyKey(property)) {
+        throw new BadArgumentException("Invalid per-table property: " + property, fullCommand,
+            fullCommand.indexOf(property));
+      }
+    }
+    if (!Property.isValidZooPropertyKey(property)) {
```

Comment:
Should this be checked first? It does not really matter if it is in the correct format or not if it can never be change in ZooKeeper.

Reply:
Makes sense to me, I'll move the code up.